autorosters/autorosters_main.py

```diff
@@ -0,0 +1,270 @@
+                self.match_data[scoreboard["MatchId"]]["games"][scoreboard["GameId"]]["sg_data"] = {
+                    "team1": scoreboard["Team1"],
+                    "team2": scoreboard["Team2"],
+                    "players": {}}
+            if scoreboard["Link"] not in player_ids_cache:
+                player_id = self.get_player_id(scoreboard["Link"])
+                player_ids_cache[scoreboard["Link"]] = player_id
+            player_page = player_ids_cache[scoreboard["Link"]]
+            game_id = scoreboard["GameId"]
+            match_id = scoreboard["MatchId"]
+            self.match_data[match_id]["games"][game_id]["sg_data"]["players"][player_page] = \
+                {"role": scoreboard["IngameRole"],
+                 "team": scoreboard["Team"],
+                 "link": player_page}
+
+    def get_players_roles_data(self):
+        for team, team_data in self.rosters_data.items():
+            for player, player_data in team_data["players"].items():
+                rolesn = len(self.rosters_data[team]["players"][player]["roles"])
+                self.rosters_data[team]["players"][player]["roles_data"]["roles"] = rolesn
+                for i, role in enumerate(self.rosters_data[team]["players"][player]["roles"]):
+                    rolen = f"role{i + 1}"
+                    rolen_short = f"r{i + 1}"
+                    self.rosters_data[team]["players"][player]["roles_data"][rolen] = role
+                    self.rosters_data[team]["players"][player]["games_by_role"][rolen_short] = ""
+
+    def initialize_roster_data(self):
+        for match in self.match_data.values():
+            for game in match["games"].values():
+                if game.get("sg_data"):
+                    for player in game["sg_data"]["players"].values():
+                        team = self.alt_teamnames[player["team"]]
+                        if team not in self.rosters_data.keys():
+                            self.rosters_data[team] = {"players": {}, "teamsvs": []}
+                        if player["link"] not in self.rosters_data[team]["players"].keys():
+                            self.rosters_data[team]["players"][player["link"]] = {"roles": [], "roles_data": {},
+                                                                                  "games_by_role": {}}
+                        if player["role"] not in self.rosters_data[team]["players"][player["link"]]["roles"]:
+                            self.rosters_data[team]["players"][player["link"]]["roles"].append(player["role"])
+        self.get_players_roles_data()
+
+    @staticmethod
+    def get_where_player_data(rosters_data):
+        players = []
+
+        where = "PR.AllName IN ("
+        for team in rosters_data.values():
+            for player in team["players"].keys():
+                if player not in players:
+                    players.append(player)
+                    where += f'"{player}", '
+        where = where[:-2] + ")"
```

Comment:
try and make a single fstring here `where = f"PR.AllName IN ({', '.join(players)})"` or something